packages/account/src/Sections/Security/SelfExclusion/self-exclusion.jsx

```diff
@@ -312,18 +298,46 @@ class SelfExclusion extends React.Component {
         this.populateExclusionResponse(get_self_exclusion_response);
     };
 
+    getLimits = async () => {
+        await WS.wait('get_account_status');
+    };
+
+    getMaxLength = field => {
+        const { account_limits, currency, is_cr } = this.props;
+
+        const decimals_length = getDecimalPlaces(currency);
+        const isIntegerField = value => /session_duration_limit|max_open_bets/.test(value);
+        const getLength = value =>
+            value.toString().length + (isIntegerField(field) || decimals_length === 0 ? 0 : decimals_length + 1); // add 1 to allow typing dot
+
+        if (/max_open_bets/.test(field) && account_limits.open_positions)
+            return getLength(account_limits.open_positions);
+
+        if (/max_balance/.test(field) && account_limits.account_balance)
+            return getLength(account_limits.account_balance);
+
+        if (!this.state.self_exclusions[field] || is_cr) {
+            if (/max_open_bets/.test(field)) return 9; // TODO: remove when the error is fixed on BE
+            return getLength(this.exclusion_fields_settings.max_number);
+        }
+
+        return getLength(this.state.self_exclusions[field]);
+    };
+
     componentDidMount() {
         const { is_virtual } = this.props;
         if (is_virtual) {
             this.setState({ is_loading: false });
         } else {
             this.getSelfExclusion();
+            this.getLimits();
         }
     }
     componentDidUpdate(prev_props) {
         if (prev_props.is_switching !== this.props.is_switching) {
             this.resetState();
             this.getSelfExclusion();
+            this.getLimits();
```

Comment:
calling `getLimits` here doesn't do anything. `getLimits` is an async function and we do await here.
```
getLimits = async () => {
     await WS.wait('get_account_status');
};
 ```
if we want `account_limits` to be populated before rendering, can we show loading until it is populated?

Reply:
As discussed previously, I reverted this function back to API call because after we update the limits, the values in client storage are not updated, therefore, unfortunately, we are not able to recalculate our validation for the given fields